pandas/core/arrays/interval.py

```diff
@@ -777,6 +777,17 @@ def take(self, indices, allow_fill=False, fill_value=None, axis=None,
 
         return self._shallow_copy(left_take, right_take)
 
+    def where(self, cond, other):
+        if is_scalar(other) and isna(other):
+            lother = rother = other
```

Comment:
To be safe, I think this should be `lother = rother = self.left._na_value` to ensure that we're filling `left`/`right` with the correct NA value.  If we use `left/right.where` instead of `np.where` this _should_ be handled automatically iirc, so could maybe just do that instead.